File: LayoutTests/http/tests/multipart/resources/multipart-wait-before-boundary.php
<?php
    if (isset($_GET['done'])) {
      header("Content-Type: text/html");
      echo("<script>parent.success()</script>");
      exit(0);
    }

    $boundary = "cutHere";

    function sendHeader()
    {
        global $boundary;

        echo("--$boundary\r\n");
        echo("Content-Type: text/html\r\n\r\n");
        flush();
    }

    header("Content-Type: multipart/x-mixed-replace; boundary=$boundary");

    // generate some padding to work around CFNetwork handling of multipart data
    $padding = "aa";
    for ($i = 0; $i < 10; $i++) {
      $padding .= $padding;
    }

    sendHeader();
    ob_end_flush();
    echo("test html\n");
    echo("<!-- $padding -->");
    flush();
    sendHeader();
    echo("second html");
    echo("<script>parent.childLoaded()</script>");
    echo("<!-- $padding -->");
    flush();
    sendHeader();
    echo("third html");
    echo("<!-- $padding -->");
    flush();
    usleep(30 * 1000000);
?>

File: LayoutTests/http/tests/multipart/resources/multipart-wait-before-boundary.py
#!/usr/bin/env python3

import os
import sys
import time
from urllib.parse import parse_qs

done = parse_qs(os.environ.get('QUERY_STRING', ''), keep_blank_values=True).get('done', [None])[0]

if done is not None:
    sys.stdout.buffer.write(
        b'Content-Type: text/html\r\n\r\n'
        b'<script>parent.success()</script>'
    )
    sys.exit(0)

boundary = 'cutHere'
padding = 'a' * 2048
def sendHeader():
    sys.stdout.buffer.write(
        '--{}\r\n'
        'Content-Type: text/html\r\n\r\n'.format(boundary).encode()
    )
    sys.stdout.flush()


sys.stdout.buffer.write('Content-Type: multipart/x-mixed-replace; boundary={}\r\n\r\n'.format(boundary).encode())

sendHeader()
sys.stdout.buffer.write(
    'test html\n'
    '<!-- {} -->'.format(padding).encode()
)
sys.stdout.flush()

sendHeader()
sys.stdout.buffer.write(
    'second html\n'
    '<script>parent.childLoaded()</script>'
    '<!-- {} -->'.format(padding).encode()
)
sys.stdout.flush()

sendHeader()
sys.stdout.buffer.write(
    'third html\n'
    '<!-- {} -->'.format(padding).encode()
)
sys.stdout.flush()

time.sleep(20)
